explorer/src/lib/components/block-details/BlockDetails.svelte

```diff
@@ -0,0 +1,170 @@
+<svelte:options immutable={true} />
+
+<script>
+  import { mdiArrowLeft, mdiArrowRight } from "@mdi/js";
+  import { AppAnchor, ListItem } from "$lib/components";
+  import { Card, Icon, ProgressBar, Switch } from "$lib/dusk/components";
+  import { createCurrencyFormatter } from "$lib/dusk/currency";
+  import {
+    calculateAdaptiveCharCount,
+    getRelativeTimeString,
+    middleEllipsis,
+  } from "$lib/dusk/string";
+  import { onMount } from "svelte";
+  import "./BlockDetails.css";
+
+  const formatter = createCurrencyFormatter("en", "DUSK", 0);
+  const blockHeight = 34526;
+
+  /** @type {number} */
+  let screenWidth = window.innerWidth;
+
+  /** @type {boolean} */
+  let togglePayload = false;
+
+  onMount(() => {
+    const resizeObserver = new ResizeObserver((entries) => {
+      const entry = entries[0];
+
+      screenWidth = entry.contentRect.width;
+    });
+
+    resizeObserver.observe(
+      document.body.getElementsByClassName("details-list__definition")[0]
+    );
+
+    return () => resizeObserver.disconnect();
+  });
+</script>
+
+<Card className="block-details">
+  <header slot="header" class="block-details__header">
+    <h3 class="block-details__header-heading">
+      Block Details <span class="block-details__header-heading--block-height">
+        - #{formatter(blockHeight)}</span
+      >
+    </h3>
+    <button type="button" on:click={() => history.back()}>Back</button>
+  </header>
+  <dl class="block-details__list">
+    <!-- BLOCK HASH -->
+    <ListItem tooltipId="blockHash" tooltipText="some text about blockhash">
+      <svelte:fragment slot="term">block hash</svelte:fragment>
+      <svelte:fragment slot="definition"
+        >{middleEllipsis(
+          "0x85a89a302b359c6729c538c92c46e3eb57a0765756be36e2411f4f38a9a45618",
+          calculateAdaptiveCharCount(screenWidth, 320, 1920, 14, 66)
+        )}</svelte:fragment
+      >
+    </ListItem>
+
+    <!-- HEIGHT -->
+    <ListItem tooltipId="height" tooltipText="some text about height">
+      <svelte:fragment slot="term">height</svelte:fragment>
+      <svelte:fragment slot="definition">
+        <AppAnchor
+          className="block-details__list-anchor"
+          href="/blocks/block?id=#"
+        >
+          <Icon path={mdiArrowLeft} />
+        </AppAnchor>
+        {formatter(blockHeight)}
+        <AppAnchor
+          className="block-details__list-anchor"
+          href="/blocks/block?id=#"
+        >
+          <Icon path={mdiArrowRight} />
+        </AppAnchor>
+      </svelte:fragment>
+    </ListItem>
+
+    <!-- TIMESTAMP -->
+    <ListItem tooltipId="timestamp" tooltipText="some text about height">
+      <svelte:fragment slot="term">timestamp</svelte:fragment>
+      <span class="block-details__list-timestamp" slot="definition">
+        {getRelativeTimeString(
+          new Date("2024-04-10 10:22:54 +0000 UTC"),
```

Comment:
You can change the date to `2024-04-10T10:22:54+00:00` so as to fix the `number argument must be finite` error.

Reply:
There's an issue here as the date string I used is what is provided by the API, I think I'm going to write a helper function to format the string.

Reply:
I see. Do we know what the consumers of the API are? If it's just the Explorer, perhaps we could tweak it on the service side, instead of creating a helper on the client?

Reply:
This is an old topic that was revised once, as the original time string was in a stranger non-ISO format.
Not sure if at this point it should be treated in the service as only Safari has a problem.